db/versions/0091-migration.sql

```diff
@@ -0,0 +1,166 @@
+  );
+
+  -- migrate data to deadline queue
+  INSERT INTO queue_task_deadlines
+    (task_group_id, task_id, scheduler_id, created, deadline, visible, pop_receipt)
+  SELECT
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'taskGroupId',
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'taskId',
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'schedulerId',
+    inserted,
+    CAST(convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'deadline' AS timestamp with time zone),
+    visible, -- when this message becomes visible, it means it hits deadline
+    pop_receipt
+  FROM azure_queue_messages
+  WHERE queue_name = 'deadline-queue'
+  AND expires > now();
+
+  CREATE INDEX queue_task_deadline_idx ON queue_task_deadlines (task_id);
+  CREATE INDEX queue_task_deadline_vis_idx ON queue_task_deadlines (visible);
+
+  -- Resolved Tasks
+  -- queries: what tasks are not scheduled for given task_group_id/scheduler_id
+  -- purpose: not much since this will have short-lived data
+  CREATE TABLE queue_resolved_tasks (
+    task_group_id text not null,
+    task_id text not null,
+    scheduler_id text not null,
+    resolution text not null,
+    resolved timestamptz not null,
+    visible timestamptz not null,
+    pop_receipt uuid null
+  );
+
+  -- migrate data to resolved queue
+  INSERT INTO
+    queue_resolved_tasks (task_id, task_group_id, scheduler_id, resolution, resolved, visible, pop_receipt)
+  SELECT
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'taskId',
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'taskGroupId',
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'schedulerId',
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'resolution',
+    inserted,
+    visible,
+    pop_receipt
+  FROM azure_queue_messages
+  WHERE queue_name = 'resolved-queue'
+  AND expires > now();
+
+  CREATE INDEX queue_resolved_task_idx ON queue_resolved_tasks (task_id);
+  CREATE INDEX queue_resolved_task_vis_idx ON queue_resolved_tasks (visible);
+
+  -- Claimed Tasks
+  -- queries: what tasks are running and waiting to be reclaimed or resolved
+  CREATE TABLE queue_claimed_tasks (
+    task_id text not null,
+    run_id integer not null,
+    task_queue_id text not null,
+    worker_group text not null,
+    worker_id text not null,
+    claimed timestamptz not null,
+    taken_until timestamptz not null,
+    visible timestamptz not null,
+    pop_receipt uuid null
+  );
+
+  -- migrate data to resolved queue
+  INSERT INTO
+    queue_claimed_tasks (task_id, run_id, task_queue_id, worker_group, worker_id, claimed, taken_until, visible, pop_receipt)
+  SELECT
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'taskId',
+    CAST(convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'runId' AS INTEGER),
+    'unknown/tq',
+    'unknonwn-wg',
+    'unknonwn-wi',
```

Comment:
There are typos in the last two of these (a typo I make all the time!)

It would probably slow down the migration to do a join with the `tasks` table to fetch these values, and I suppose wouldn't add much value. Maybe add a comment describing the temporary impact of using these dummy values?

Reply:
I should install spell checker probably 🤦 

I wanted to add those fields for future uses like query all tasks for a given worker or task queue.. but this also can be added later also. 
I can also just drop those columns for now, since they are not used  yet

Reply:
I figured that is what the fields were for. If they will be populated as new tasks are added, then it makes sense to include them now -- then by the time there's functionality to _read_the columns, they will have valid data in them.